app/src/main/java/com/termux/app/terminal/TermuxTerminalViewClient.java

```diff
@@ -0,0 +1,672 @@
+package com.termux.app.terminal;
+
+import android.annotation.SuppressLint;
+import android.app.AlertDialog;
+import android.content.ActivityNotFoundException;
+import android.content.ClipData;
+import android.content.ClipboardManager;
+import android.content.Context;
+import android.content.Intent;
+import android.media.AudioManager;
+import android.net.Uri;
+import android.text.TextUtils;
+import android.view.Gravity;
+import android.view.InputDevice;
+import android.view.KeyEvent;
+import android.view.MotionEvent;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.Toast;
+
+import com.termux.R;
+import com.termux.app.TermuxActivity;
+import com.termux.shared.shell.ShellUtils;
+import com.termux.shared.terminal.TermuxTerminalViewClientBase;
+import com.termux.shared.termux.TermuxConstants;
+import com.termux.app.activities.ReportActivity;
+import com.termux.app.models.ReportInfo;
+import com.termux.app.models.UserAction;
+import com.termux.app.terminal.io.KeyboardShortcut;
+import com.termux.app.terminal.io.extrakeys.ExtraKeysView;
+import com.termux.shared.settings.properties.TermuxPropertyConstants;
+import com.termux.shared.data.DataUtils;
+import com.termux.shared.logger.Logger;
+import com.termux.shared.markdown.MarkdownUtils;
+import com.termux.shared.termux.TermuxUtils;
+import com.termux.shared.view.KeyboardUtils;
+import com.termux.terminal.KeyHandler;
+import com.termux.terminal.TerminalEmulator;
+import com.termux.terminal.TerminalSession;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+
+import androidx.drawerlayout.widget.DrawerLayout;
+
+public class TermuxTerminalViewClient extends TermuxTerminalViewClientBase {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 58 other classes. By comparison, 99% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.